File: app/src/main/java/com/example/kotlinhomework/SecondActivity.java
package com.example.kotlinhomework;

import android.content.Intent;
import android.os.Bundle;
import android.text.Editable;
import android.text.TextUtils;
import android.text.TextWatcher;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;

import androidx.appcompat.app.AppCompatActivity;

public class SecondActivity extends AppCompatActivity {

    private EditText editText;
    private Button button;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_second);

        editText = findViewById(R.id.editTextTextPersonName);
        button = findViewById(R.id.buttonName);
        button.setEnabled(false);
        editText.addTextChangedListener(new TextWatcher() {
            @Override
            public void beforeTextChanged(CharSequence charSequence, int i, int i1, int i2) {
                // Nothing to do
            }

            @Override
            public void onTextChanged(CharSequence charSequence, int i, int i1, int i2) {
                // Nothing to do
            }

            @Override
            public void afterTextChanged(Editable editable) {
                button.setEnabled(!TextUtils.isEmpty(editable));
            }

        });
        button.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                Intent intent = new Intent();
                intent.putExtra(NAME_KEY, editText.getText().toString());
                setResult(RESULT_OK, intent);
                finish();
            }
        });
    }

    public static final String NAME_KEY = "com.example.kotlinhomework.SecondActivity.NAME_KEY";
    public static final int GET_NAME_REQUEST_CODE = 1234;
}

File: app/src/main/java/com/example/kotlinhomework/SecondActivity.kt
package com.example.kotlinhomework

import android.content.Intent
import android.os.Bundle
import android.text.Editable
import android.text.TextUtils
import android.text.TextWatcher
import android.view.View
import android.widget.Button
import android.widget.EditText
import androidx.appcompat.app.AppCompatActivity

class SecondActivity : AppCompatActivity() {
    private var editText: EditText? = null
    private var button: Button? = null
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_second)
        editText = findViewById(R.id.editTextTextPersonName)
        button = findViewById(R.id.buttonName)
        button?.isEnabled = false
        editText?.addTextChangedListener(object : TextWatcher {
            override fun beforeTextChanged(charSequence: CharSequence, i: Int, i1: Int, i2: Int) {
                // Nothing to do
            }

            override fun onTextChanged(charSequence: CharSequence, i: Int, i1: Int, i2: Int) {
                // Nothing to do
            }

            override fun afterTextChanged(editable: Editable) {
                button?.isEnabled = !TextUtils.isEmpty(editable)
            }
        })
        button?.setOnClickListener(View.OnClickListener {
            val intent = Intent()
            intent.putExtra(NAME_KEY, editText?.text.toString())
            setResult(RESULT_OK, intent)
            finish()
        })
    }

    companion object {
        const val NAME_KEY = "com.example.kotlinhomework.SecondActivity.NAME_KEY"
        const val GET_NAME_REQUEST_CODE = 1234
    }
}